apps/mobile-wallet/src/store/addressesSlice.ts

```diff
@@ -325,24 +326,36 @@ export const makeSelectAddressesAlphAsset = () =>
     }
   })
 
-// Same as in desktop wallet
 export const makeSelectAddressesTokens = () =>
   createSelector(
-    [selectAllFungibleTokens, selectAllNFTs, makeSelectAddressesAlphAsset(), makeSelectAddresses(), selectAllPrices],
-    (fungibleTokens, nfts, alphAsset, addresses, tokenPrices): Asset[] => {
+    [
+      selectAllFungibleTokens,
+      selectAllNFTs,
+      makeSelectAddressesAlphAsset(),
+      makeSelectAddresses(),
+      selectAllPrices,
+      (_state: RootState, _addressHash?: AddressHash | AddressHash[], filterHiddenTokens?: boolean) =>
+        filterHiddenTokens ?? false,
+      selectHiddenAssetsIds
+    ],
+    (fungibleTokens, nfts, alphAsset, addresses, tokenPrices, filterHiddenTokens, hiddenAssetIds): Asset[] => {
+      // Your existing logic
       const tokenBalances = getAddressesTokenBalances(addresses)
 
-      if (alphAsset.balance > BigInt(0)) tokenBalances.push(alphAsset)
+      if (alphAsset.balance > BigInt(0)) {
+        tokenBalances.push(alphAsset)
+      }
 
-      const tokens = calculateAssetsData(tokenBalances, fungibleTokens, nfts, tokenPrices)
+      const tokens = calculateAssetsData(tokenBalances, fungibleTokens, nfts, tokenPrices).filter((t) =>
+        filterHiddenTokens ? !hiddenAssetIds.includes(t.id) : true
+      )
 
       return sortAssets(tokens)
```

Comment:
Let's avoid looping over the array when it's not necessary:

```suggestion
      const tokens = calculateAssetsData(tokenBalances, fungibleTokens, nfts, tokenPrices)

      return sortAssets(filterHiddenTokens ? tokens.filter((t) => !hiddenAssetIds.includes(t.id)) : tokens)
```

Reply:
Fixed in my PR